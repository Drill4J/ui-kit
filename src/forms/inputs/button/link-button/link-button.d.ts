/// <reference types="react" />
interface Props {
    className?: string;
    children: React.ReactNode;
    onClick?: () => void;
    size?: 'large' | 'small';
    disabled?: boolean;
    'data-test'?: string;
}
export declare const LinkButton: ({ className, children, ...rest }: Props) => JSX.Element;
export {};
