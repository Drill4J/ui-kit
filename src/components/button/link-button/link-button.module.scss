@import '../../../style-guide/index';

.link-button {
  padding: 0;
  border: none;
  background-color: transparent;
  color: map-get($color-palette, blue100);
  font-family: $semibold-font;

  &:hover {
    color: map-get($color-palette, blue75);
  }

  &:active {
    color: map-get($color-palette, blue125);
  }
}
