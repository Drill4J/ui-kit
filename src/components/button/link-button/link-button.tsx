import * as React from 'react';
import { BEM } from '@redneckz/react-bem-helper';

import { Button } from '../button';
import { ButtonType } from '../button-type';

import styles from './link-button.module.scss';

interface Props extends ButtonType {
  className?: string;
  children: React.ReactNode;
  onClick?: () => void;
}

const linkButton = BEM(styles);

export const LinkButton = linkButton(({ className, children, ...rest }: Props) => (
  <Button className={className} {...rest}>
    {children}
  </Button>
));
