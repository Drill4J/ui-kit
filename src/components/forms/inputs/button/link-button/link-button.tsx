import tw, { styled } from 'twin.macro';

import { Button } from '../button';

export const LinkButton = styled(Button)`
  ${tw`
    inline-flex p-0 border-none bg-transparent text-blue-default font-bold
    hover:text-blue-medium-tint
    active:text-blue-shade
  `};
`;
