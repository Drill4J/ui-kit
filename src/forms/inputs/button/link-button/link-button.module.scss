@import '../../../../style-guide/index';

.link-button {
  display: inline-flex;
  padding: 0;
  border: none;
  background-color: transparent;
  color: map-get($primary-blue, default);
  font-family: $semibold-font;
  &:hover {
    color: map-get($primary-blue, medium-tint);
  }

  &:active {
    color: map-get($primary-blue, shade);
  }
}
