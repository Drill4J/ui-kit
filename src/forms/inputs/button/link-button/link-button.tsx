import { BEM } from '@redneckz/react-bem-helper';

import { Button } from '../button';

import styles from './link-button.module.scss';

interface Props {
  className?: string;
  children: React.ReactNode;
  onClick?: () => void;
  size?: 'large' | 'small';
  disabled?: boolean;
  'data-test'?: string;
}

const linkButton = BEM(styles);

export const LinkButton = linkButton(({ className, children, ...rest }: Props) => (
  <Button className={className} {...rest}>
    {children}
  </Button>
));
