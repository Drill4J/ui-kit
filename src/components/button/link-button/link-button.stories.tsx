import React from 'react';
import { storiesOf } from '@storybook/react';

import { LinkButton } from './link-button';

storiesOf('LinkButton', module)
  .add('Primary large', () => <LinkButton size="large">Primary large</LinkButton>)
  .add('Secondary small', () => <LinkButton size="small">Secondary small</LinkButton>);
