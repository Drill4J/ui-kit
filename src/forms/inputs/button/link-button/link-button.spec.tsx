import React from 'react';
import renderer from 'react-test-renderer';

import { LinkButton } from './link-button';

it('should match snapshot', () => {
  const tree = renderer.create(<LinkButton size="large">Link button</LinkButton>).toJSON();

  expect(tree).toMatchSnapshot();
});
