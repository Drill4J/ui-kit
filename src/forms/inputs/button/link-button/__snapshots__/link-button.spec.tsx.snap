// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<button
  className="sc-bdvvaa sc-gsDJrp dEcWDh dcPFaG"
  size="large"
>
  Link button
</button>
`;
